src/dist/replication/lib/bulk_load/replica_bulk_loader.cpp

```diff
@@ -380,6 +418,35 @@ void replica_bulk_loader::check_download_finish()
     }
 }
 
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_bulk_loader::start_ingestion()
+{
+    _status = bulk_load_status::BLS_INGESTING;
+    // TODO(heyuchen): add perf-counter
+    if (status() == partition_status::PS_PRIMARY) {
+        _replica->_primary_states.ingestion_is_empty_prepare_sent = false;
+    }
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_bulk_loader::check_ingestion_finish()
```

Comment:
I understood the method is `check whether the ingetion is finished`, but the logic seem mean `gurantee it can be finished`?

Reply:
Well, it is depends on when ingestion is finished, in my view, this function means check whether primary replica's ingestion_status is ingestion_succeed, besides, if it has not sent empty prepare to secondaries, sends empty prepare to make secondaries commit this mutation, I think it is to guarantee data integrity. Actually, when primary sends empty prepare to secondaries, ingestion is not finished. All replicas' ingestion_status are ingestion_succeed means ingestion finished.